openff/toolkit/tests/test_topology.py

```diff
@@ -820,6 +825,146 @@ def test_to_file_multi_molecule_different_order(self):
+    def test_to_file_automatic_positions(self):
+        """
+        Checks that to_file can take positions from the topology
+        """
+        from io import StringIO
+
+        from openff.toolkit.topology import Molecule, Topology
+
+        topology = Topology()
+        mol = Molecule.from_pdb_and_smiles(
+            get_data_file_path("systems/test_systems/1_ethanol.pdb"), "CCO"
+        )
+        topology.add_molecule(mol)
+
+        count = 1
+        with StringIO() as iofile:
+            topology.to_file(iofile)
+            data = iofile.getvalue().splitlines()
+            for line in data:
+                if line.startswith("HETATM") and count == 1:
+                    count = count + 1
+                    coord = line.split()[-6]
+        assert coord == "10.172"
+
+    def test_to_file_ensure_uniqueness_true(self):
+        """
+        Checks that ensure_unique_atom_names=True provides per-molecule unique atom names
+        """
+        from io import StringIO
+
+        from openff.toolkit.topology import Molecule, Topology
+
+        mol = Molecule.from_polymer_pdb(
+            get_data_file_path("proteins/MainChain_ALA_ALA.pdb")
+        )
+        topology = Topology.from_molecules([mol])
+
+        with StringIO() as iofile:
+            topology.to_file(iofile, ensure_unique_atom_names=True)
+            data = iofile.getvalue().splitlines()
+
+        atom_names = []
+        for line in data:
+            if line.startswith("ATOM") or line.startswith("HETATM"):
+                atom_name = line[12:16].strip()
+                atom_names.append(atom_name)
+        assert len(atom_names) == len(set(atom_names))
+
+    def test_to_file_ensure_uniqueness_false(self):
+        """
+        Checks that ensure_unique_atom_names=False preserves atom names
+        """
+        from io import StringIO
+
+        from openff.toolkit.topology import Molecule, Topology
+
+        mol = Molecule.from_polymer_pdb(
+            get_data_file_path("proteins/MainChain_ALA_ALA.pdb")
+        )
+        topology = Topology.from_molecules([mol])
+        atom_names = set([atom.name for atom in topology.atoms])
+        assert None not in atom_names, "All input atoms must be named"
+
+        with StringIO() as iofile:
+            topology.to_file(iofile, ensure_unique_atom_names=False)
+            data = iofile.getvalue().splitlines()
+
+        for line in data:
+            if line.startswith("ATOM") or line.startswith("HETATM"):
+                atom_name = line[12:16].strip()
+                assert atom_name in atom_names
+
+    def test_to_file_ensure_uniqueness_residues(self):
+        """
+        Checks that ensure_unique_atom_names="residues" provides per-residue unique atom names
+        """
+        from io import StringIO
+
+        from openff.toolkit.topology import Molecule, Topology
+
+        mol = Molecule.from_polymer_pdb(
+            get_data_file_path("proteins/MainChain_ALA_ALA.pdb")
```

Comment:
(not blocking) The atom names in this file are already unique, so we wouldn't expect any atom renaming to happen in this test. It's still a useful test as-is though.

Reply:
Yeah I guess the point of this test is just to make sure the argument is getting passed along to `to_openmm`. The exact behavior is tested in the tests for that method.